docs/guides/interfaces.rst

```diff
@@ -1,10 +1,89 @@
-.. _interfaces_guide:
+.. _interfaces:
+========================
+Interfaces
+========================
 
-===================
-Interfaces guide
-===================
 
-This guide is currently a work in progress and only contains bullet points for now.
+One of the biggest strengths of Orquestra is its modularity. Integrating new backends, optimizers, ansatzes, etc. does not require changing the core code - it requires only creating a new module that conforms to existing interfaces and therefore can be used across the whole platform. Using our interfaces help you add whatever building blocks you need for your project and ensure that they will work well with other existing elements.
 
-Cost Functions in VQAs
-======================
\ No newline at end of file
+What is an interface?
+=====================
+
+Before we get into the details, let's explain what we mean by "interfaces" in this context. We use this word as a shorthand for "a way to implement certain concepts in such a way, that as long as two implementations conform to a particular interface, they can be swapped and the program should still work correctly". Obviously, various implementations might work differently - e.g.: one simulator might be faster than the other - but they should be undistinguishable when it comes to inputs, outputs and how they are being used.
+
+
+
+Benefits of interfaces
+======================
+
+Interface offers the ability to define behaviour that can be implemented by unrelated classes without forcing them to share a common class hierarchy. This results in code flexibility and faster development with easier testing options. Other benefits include:
+
+* Interoperability
+* Modularity
+* Dependencies decoupling
+
+
+Interface implementation
+========================
+
+Currently interfaces in Orquestra are defined in the following 3 main libraries:
+
+* `orquestra-quantum <https://github.com/zapatacomputing/orquestra-quantum>`_
+* `orquestra-opt <https://github.com/zapatacomputing/orquestra-opt>`_
+* `orquestra-vqa <https://github.com/zapatacomputing/orquestra-vqa>`_
+
+The interfaces are located in the ``api`` directory within each packages. They are implemented either as an ``abstract class`` or ``protocols``. `Abstract classes <https://docs.python.org/3/library/abc.html>`_ are classes that contain one or more abstract method. However, they do not contain any implementation. Subclasses are required to provide implementations for abstract methods. Abstract classes can also be considered blueprint classes. 
+
+However, using abstract classes requires using inheritance, which sometimes is undesirable. Therefore, we also make use of `Protocols <https://peps.python.org/pep-0544/>`_ . The main differences are that we don't have to explicitly express that particular implementation conforms to a protocol and that protocols work not just with classes, but also functions. 
+
+Interfaces provided by Orquestra are:
```

Comment:
I think that the following sounds better, but it is very subjective so feel free to ignore it:
```suggestion
Orquestra provides the following interfaces:
```